flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/serialization/SpanningRecordSerializerTest.java

```diff
@@ -106,17 +102,19 @@ public boolean equals(Object obj) {
 			}
 		};
 
-		RecordSerializer.SerializationResult result = serializer.addRecord(emptyRecord);
-		Assert.assertEquals(RecordSerializer.SerializationResult.FULL_RECORD, result);
+		serializer.serializeRecord(emptyRecord);
+		Assert.assertEquals(RecordSerializer.SerializationResult.FULL_RECORD, serializer.copyToBufferBuilder(bufferBuilder1));
 
-		result = serializer.addRecord(emptyRecord);
-		Assert.assertEquals(RecordSerializer.SerializationResult.FULL_RECORD, result);
+		serializer.serializeRecord(emptyRecord);
+		Assert.assertEquals(RecordSerializer.SerializationResult.FULL_RECORD, serializer.copyToBufferBuilder(bufferBuilder1));
 
-		result = serializer.addRecord(emptyRecord);
-		Assert.assertEquals(RecordSerializer.SerializationResult.PARTIAL_RECORD_MEMORY_SEGMENT_FULL, result);
+		serializer.serializeRecord(emptyRecord);
```

Comment:
why don't you use `serializer.reset()` here? (serialize only once as in production code)